distributed/worker.py

```diff
@@ -536,6 +536,7 @@ def __init__(
             ("cancelled", "resumed"): self.transition_cancelled_resumed,
             ("cancelled", "fetch"): self.transition_cancelled_fetch,
             ("cancelled", "released"): self.transition_cancelled_released,
+            ("cancelled", "missing"): self.transition_cancelled_released,
```

Comment:
For those transition functions that are used in many places we're typically using something like `transition_generic_released`.
I don't have a good idea on how to name `transition_cancelled_released` to be more explicit (it is not very generic just because it is used in two places). I'm fine either way. It's very easy to look up for which transitions they are used.